api/client/client.go

```diff
@@ -0,0 +1,249 @@
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
+	"github.com/ethereum/go-ethereum/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type BlobEncodingVersion byte
+
+var NoIFFT BlobEncodingVersion = 0x00
+
+type IEigenDAClient interface {
+	GetBlob(ctx context.Context, BatchHeaderHash []byte, BlobIndex uint32) ([]byte, error)
+	PutBlob(ctx context.Context, txData []byte) (*Cert, error)
+}
+
+type EigenDAClient struct {
+	Config
+
+	Log log.Logger
+
+	client disperser.DisperserClient
+
+	signer *auth.LocalBlobRequestSigner
+}
+
+var _ IEigenDAClient = EigenDAClient{}
+
+func NewEigenDAClient(log log.Logger, config Config) (*EigenDAClient, error) {
+	err := config.Check()
+	if err != nil {
+		return nil, err
+	}
+	client, err := NewDisperserClient(config.RPC)
+	if err != nil {
+		return nil, fmt.Errorf("failed to instatiated EigenDA GRPC client: %w", err)
+	}
+	return &EigenDAClient{
+		Log:    log,
+		Config: config,
+		client: client,
+		signer: auth.NewLocalBlobRequestSigner(config.SignerPrivateKeyHex),
+	}, nil
+}
+
+func NewDisperserClient(rpc string) (disperser.DisperserClient, error) {
+	config := &tls.Config{}
+	credential := credentials.NewTLS(config)
+	dialOptions := []grpc.DialOption{grpc.WithTransportCredentials(credential)}
+	conn, err := grpc.Dial(rpc, dialOptions...)
+	if err != nil {
+		return nil, err
+	}
+	client := disperser.NewDisperserClient(conn)
+	return client, nil
+}
+
+func (m EigenDAClient) GetBlob(ctx context.Context, BatchHeaderHash []byte, BlobIndex uint32) ([]byte, error) {
+	reply, err := m.client.RetrieveBlob(ctx, &disperser.RetrieveBlobRequest{
+		BatchHeaderHash: BatchHeaderHash,
+		BlobIndex:       BlobIndex,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// decode modulo bn254
+	decodedData := codec.RemoveEmptyByteFromPaddedBytes(reply.Data)
+
+	// Return exact data with buffer removed
+	reader := bytes.NewReader(decodedData)
+
+	// read version byte, we will not use it for now since there is only one version
+	_, err = reader.ReadByte()
+	if err != nil {
+		return nil, fmt.Errorf("EigenDA client failed to read version byte")
+	}
+
+	// read length uvarint
+	length, err := binary.ReadUvarint(reader)
+	if err != nil {
+		return nil, fmt.Errorf("EigenDA client failed to decode length uvarint prefix")
+	}
+	data := make([]byte, length)
+	n, err := reader.Read(data)
+	if err != nil {
+		return nil, fmt.Errorf("EigenDA client failed to copy unpadded data into final buffer")
+	}
+	if uint64(n) != length {
+		return nil, fmt.Errorf("EigenDA client failed, data length does not match length prefix")
+	}
+
+	return data, nil
+}
+
+func (m EigenDAClient) PutBlob(ctx context.Context, data []byte) (*Cert, error) {
+	m.Log.Info("Attempting to disperse blob to EigenDA")
+
+	// encode current blob encoding version byte
+	data = append([]byte{byte(NoIFFT)}, data...)
+
+	// encode data length
+	data = append(ConvertIntToVarUInt(len(data)), data...)
```

Comment:
How about adding a version as well ? To start with, version 0, or something similar, so that in the future, if we add more context like this, we can bump the version to know which deserialization would need to happen.

Reply:
It was my intention that the version byte written on line 115 would describe the encoding spec for the entire blob. Is there a benefit to having distinct version bytes for different subsections of the blob's encoding?

Reply:
I think we can bump the main version incase something changes in the subsections as well, so should be fine.